src/Mod/TechDraw/App/DrawViewPart.cpp

```diff
@@ -225,25 +221,23 @@ TopoDS_Shape DrawViewPart::getSourceShapeFused(void) const
             Base::Console().Error("Error: DVP::getSourceShape - No Source(s) linked. - %s\n",
                                   getNameInDocument());
         }
-    } else {
-        result = ShapeExtractor::getShapesFused(links);
+        return TopoDS_Shape();
     }
-    return result;
+    return ShapeExtractor::getShapesFused(links);
 }
 
 std::vector<App::DocumentObject*> DrawViewPart::getAllSources(void) const
 {
 //    Base::Console().Message("DVP::getAllSources()\n");
-    const std::vector<App::DocumentObject*> links = Source.getValues();
+    std::vector<App::DocumentObject*> links = Source.getValues();
```

Comment:
This is wrong, isn't it?

Reply:
Source.getValues says it returns const vector and XSource.getValues says it returns non-const vector.   
Why are they different? I don't know. 

We only read the links. We don't add, change or delete any entries, so const is fine, but I'm not sure const makes a big difference here since the vector doesn't have many entries, and the entries are small.

Reply:
I haven't looked at the rest of the code, but note that here a copy is made instead of referencing (`type&`). If it needs to be a copy I don't see why it should be const. If it needs to be const I don't see the need to copy. Though non const reference is possible.